curvefs/src/client/rpcclient/metacache.cpp

```diff
@@ -383,12 +372,16 @@ bool MetaCache::DoListOrCreatePartitions(
 
 void MetaCache::DoAddPartitionAndCopyset(
     const PatitionInfoList &partitionInfos,
-    const std::map<PoolIDCopysetID, CopysetInfo<MetaserverID>> &copysetMap) {
+    const std::map<PoolIDCopysetID, CopysetInfo<MetaserverID>> &copysetMap,
+    bool refresh) {
+    if (refresh) {
+        partitionInfos_.clear();
+        copysetInfoMap_.clear();
+    }
```

Comment:
it's better to rename this function to `DoAddOrReset...`, and the bool parameter should rename to `add` or `reset`.

and another suggestion is also about moving resources, the first two arguments that the caller passed can be moved into `partitionInfos_` and `copysetInfoMap_`, because they 
are local variables and then copy to partitionInfos_ and copysetInfoMap_, finally, they got destroyed.

for example,
```cpp
void DoResetOrAddCopyset(PatitionInfoList partitionInfos,
    std::map<PoolIDCopysetID, CopysetInfo<MetaserverID>> copysetMap) { // parameters are passed by value

    partitionInfos_.insert(partitionInfos_.end(), std::make_move_iterator(partitionInfos.beign()), std::make_move_iterator(partitionInfos.end()));

}
```

// in ListPartitions
```cpp
fsID_ = fsID;
PatitionInfoList partitionInfos;
std::map<PoolIDCopysetID, CopysetInfo<MetaserverID>> copysetMap;
if (!DoListOrCreatePartitions(true, &partitionInfos, &copysetMap)) {
    return false;
}

DoAddPartitionAndCopyset(std::move(partitionInfos), std::move(copysetMap), true);
```

// in CreatePartitions
```cpp
// do not move the newPartitions, because it's a output parameter
DoAddPartitionAndCopyset(*newPartitions, std::move(copysetMap), false);
```